GVFS/GVFS.Common/FileSystem/PhysicalFileSystem.cs

```diff
@@ -260,6 +256,50 @@ public bool TryCopyToTempFileAndRename(string sourcePath, string destinationPath
             }
         }
 
+        public bool TryCreateDirectory(string path, out Exception exception)
+        {
+            try
+            {
+                Directory.CreateDirectory(path);
+            }
+            catch (Exception e) when (e is IOException ||
+                                      e is UnauthorizedAccessException ||
+                                      e is ArgumentException ||
+                                      e is NotSupportedException)
+            {
+                exception = e;
+                return false;
+            }
+
+            exception = null;
+            return true;
+        }
+
+        /// <summary>
+        /// Recursively deletes a directory and all contained contents.
+        /// </summary>
+        public bool TryDeleteDirectory(string path, out Exception exception)
+        {
+            try
+            {
+                this.DeleteDirectory(path);
+            }
+            catch (DirectoryNotFoundException)
+            {
+                // Swallow the exception
+            }
+            catch (Exception e) when (e is IOException ||

```

Comment:
Following the call chain here, to `this.DeleteDirectory -> RecursiveDelete > DirectoryInfo.Delete` it appears the `DirectoryNotFoundException` could also be [thrown](https://docs.microsoft.com/en-us/dotnet/api/system.io.directoryinfo.delete?view=netframework-4.7.2).

Does this method `catch` all possible problems?

I also notice the `TryCreateDirectory` method above does not catch `PathTooLongException`.. is that something we consider valid to `catch`?

Reply:
> `DirectoryNotFound` and `PathTooLong` exceptions are subclass of the `System.IO` exception, which is caught.

Ahh, good point I had missed that!

> I asked myself the same question, though. I could be explicit about the exceptions that are caught and just add them, to remove future confusion...

It probably isn't worth it?